ssl/statem/statem_lib.c

```diff
@@ -2190,7 +2193,7 @@ int ssl_choose_server_version(SSL_CONNECTION *s, CLIENTHELLO_MSG *hello,
 
     if (suppversions->present) {
         unsigned int candidate_vers = 0;
-        const unsigned int best_vers_init = SSL_CONNECTION_IS_DTLS(s) ? UINT_MAX
+        const unsigned int best_vers_init = SSL_CONNECTION_IS_DTLS(s) ? INT_MAX
```

Comment:
Why this change?

Reply:
Because version numbers are treated as int's in `ssl_version_cmp`. So actually we should change the type here?

Reply:
Yeah - that would be ok and avoids the mismatch between using the signed max and the unsigned type.